[MASWE-0020] Weak Encryption (by @appknox) (#2910)
OWASP/owasp-mastg: demos/android/MASVS-CRYPTO/MASTG_DEMO-0016/MASTG-DEMO-0016.md

---
platform: android

> title: Weak Encryption

Reviewer: To go inline with MAST-DEMO-0018
```suggestion
title: Uses of Insecure Encryption Algorithms in Cipher with se grep
```

> title: Uses of Insecure Encryption Algorithms in Cipher with semgrep

Author: - do we need to create new Demo file for  `title: Uses of Insecure Encryption Modes in Cipher with semgrep` or make changes to the existing one.
- The `test:MASTG-TEST-0231` point to [Reference to logging of APIs](https://github.com/OWASP/owasp-mastg/blob/master/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0231.md).

> title: Uses of Insecure Encryption Algorithms in Cipher with semgrep

Reviewer: The ID of the "Weak Encryption Modes" has changed to MASTG-TEST-0232  https://mas.owasp.org/MASTG/tests-beta/android/MASVS-CRYPTO/MASTG-TEST-0232/

You may add a new demo pointing to that test or do it on a separate PR.

You can keep this demo with the changes indicated below.

id: MASTG-DEMO-0016
code: [java]

> test: MASTG-TEST-0211

Reviewer: This can point to the existing test 
```suggestion
test: MASTG-TEST-0221
```

---

### Sample

{{ MastgTest.kt # MastgTest_reversed.java }}

### Steps

Let's run our @MASTG-TOOL-0110 rule against the sample code.

{{ ../../../../rules/mastg-android-weak-encryption.yaml }}

{{ run.sh }}

### Observation

> The rule has identified five instances in the code file where an insecure encryption is used. The specified line numbers can be located in the original code for further investigation and remediation.

Reviewer: ```suggestion
The rule has identified five instances in the code file where insecure encryption algorithms are used.
```

---
platform: android
title: Uses of Insecure Encryption Modes in Cipher with semgrep
id: MASTG-DEMO-0016

> code: [java]

Reviewer: ```suggestion
code: [kotlin]
```

test: MASTG-TEST-0221
---

> ### Sample

Reviewer: Add short sample description

{{ MastgTest.kt # MastgTest_reversed.java }}

### Steps

Let's run our @MASTG-TOOL-0110 rule against the sample code.

{{ ../../../../rules/weak-encryption-modes.yaml }}

{{ run.sh }}

### Observation

> The rule has identified five instances in the code file where insecure encryption algorithms are used.

Reviewer: ```suggestion
The rule has identified five instances in the code where insecure encryption modes are used.
```

{{ output.txt }}

### Evaluation

> Review each of the reported instances. The following configuration modes will implies the usage of insecure AES/ECB:

Reviewer: ```suggestion
The test fails since the output contains several instances of the ECB mode of AES:
```
